pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2188,14 +2188,25 @@ func DomainCreate(status types.DomainStatus) (int, string, error) {
 		// Use rkt tool
 		// get the rkt image hash for this file; if we do not have it in the rkt cache,
 		// convert it
-		// XXX we assume a container has one image. XXX do we check somewhere?
-		ociFilename, err := utils.VerifiedImageFileLocation(status.DiskStatusList[0].ImageSha256)
+		var containerImageSha256 []string
+		for _, ds := range status.DiskStatusList {
+			if ds.Format == zconfig.Format_CONTAINER {
+				containerImageSha256 = append(containerImageSha256, ds.ImageSha256)
```

Comment:
This append is odd. It looks like you are just counting the number of Fomat_CONTAINER, and that is best done with an int e.g., numContainers := 0 before the loop and incremented here.

Reply:
However, you are already inside a loop over range config.DiskConfigList, thus it might make sense to check for the number of containers separately, e.g., by adding a loop over range config.DiskConfigList at the top of this function or in some new validateConfig() called from our caller.

Reply:
Yes, I am counting the number of container disks in the drives. It is not necessary that container disk will always be present at index 0 in the drive that's why I am appending the image SHA.

Reply:
Yeah, I can add this check-in separate function and will call that from a caller. I will update my PR.

Reply:
> Yes, I am counting the number of container disks in the drives. It is not necessary that container disk will always be present at index 0 in the drive that's why I am appending the image SHA.

Implementing a counter by appending sha strings qualifies for the obfuscated programming contest ;-)